contracts/core/src/token/OmniGasPump.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.24;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol";
+import { XAppUpgradeable } from "src/pkg/XAppUpgradeable.sol";
+import { FeeOracleV1 } from "src/xchain/FeeOracleV1.sol";
+import { ConfLevel } from "src/libraries/ConfLevel.sol";
+import { OmniXFund } from "./OmniXFund.sol";
+
+/**
+ * @title OmniGasPump
+ * @notice A unidirectional cross-chain gas exchange, allowing users to swap native ETH for native OMNI.[]
+ */
+contract OmniGasPump is XAppUpgradeable, OwnableUpgradeable, PausableUpgradeable {
+    event MaxSwapSet(uint256 max);
+    event FundSet(address fund);
+    event PctCutSet(uint256 pct);
+    event Refilled(address indexed recipient, uint256 paid, uint256 fee, uint256 cut, uint256 amtOMNI);
+
+    /// @notice Gas limit passed to OmniXFund.settleUp xcall
+    uint64 internal constant SETTLE_GAS = 100_000;
+
+    /// @notice Denominator for pct cut
+    uint256 internal constant PCT_DENOM = 1000;
+
+    /// @notice Address of OmniXFund on Omni
+    address public fund;
+
+    /// @notice Max amt (in native token) that can be swapped in a single tx
+    uint256 public maxSwap;
+
+    /// @notice Percentage cut (over PCT_DENOM) taken by this contract for each swap
+    ///     Used to disencentivize spamming
+    uint256 public pctCut;
```

Comment:
don't love this name, it's not very descriptive imo

what about deduction or fee?

Reply:
- fee is tricky cause we use if for xcall stuff
- IMO deduction means the same thing as cut

I agree I don't  _love_ the name. But it's not really user facing. 

Maybe toll?

Reply:
Toll is great